### PR TITLE
refactor: arrow conversion function

### PR DESCRIPTION
**Pull request by chloeminkyung**
remove duplicate `map_value_to_dozer_field()` function

### CODE REVIEW

File: dozer-types/src/arrow_types/from_arrow.rs [hunk 1]
```diff
@@ -275,8 +296,9 @@ pub fn map_record_batch_to_dozer_records(
     for r in 0..num_rows {
         let mut values = vec![];
         for (c, x) in columns.iter().enumerate() {
+            let metadata = batch_schema.metadata();
```
Comment (chubei) [inline dozer-types/src/arrow_types/from_arrow.rs:299]: Move this call out of the loop?

File: dozer-ingestion/src/connectors/object_store/table_reader.rs [hunk 2]
```diff
@@ -81,11 +81,19 @@ impl<T: Clone + Send + Sync> TableReader<T> {
 
             for row in 0..batch.num_rows() {
                 let fields = batch
+                    .to_owned()
                     .columns()
                     .iter()
                     .enumerate()
                     .map(|(col, column)| {
-                        map_value_to_dozer_field(column, &row, resolved_schema.field(col).name())
+                        let batch_schema = batch.schema();
```
Comment (chubei) [inline dozer-ingestion/src/connectors/object_store/table_reader.rs:89]: Move this call out of the loop?

File: dozer-ingestion/src/connectors/object_store/table_reader.rs [hunk 3]
```diff
@@ -81,11 +81,19 @@ impl<T: Clone + Send + Sync> TableReader<T> {
 
             for row in 0..batch.num_rows() {
                 let fields = batch
+                    .to_owned()
```
Comment (chubei) [inline dozer-ingestion/src/connectors/object_store/table_reader.rs:84]: Why we need to clone the batch?

File: dozer-ingestion/src/connectors/delta_lake/reader.rs [hunk 4]
```diff
@@ -47,11 +47,15 @@ impl DeltaLakeReader {
         tokio::pin!(data);
         while let Some(Ok(batch)) = data.next().await {
             for row in 0..batch.num_rows() {
+                let batch_schema = batch.schema();
+                let metadata = batch_schema.metadata();
```
Comment (chubei) [inline dozer-ingestion/src/connectors/delta_lake/reader.rs:51]: Move these calls out of the loop?

**Review by chubei — APPROVED** [review]

File: dozer-types/src/arrow_types/from_arrow.rs (hunk 1 above)
Comment (chloeminkyung) [reply dozer-types/src/arrow_types/from_arrow.rs:299]: right!

File: dozer-ingestion/src/connectors/object_store/table_reader.rs (hunk 2 above)
Comment (chloeminkyung) [reply dozer-ingestion/src/connectors/object_store/table_reader.rs:89]: sure

File: dozer-ingestion/src/connectors/object_store/table_reader.rs (hunk 3 above)
Comment (chloeminkyung) [reply dozer-ingestion/src/connectors/object_store/table_reader.rs:84]: can remove now after refactor

File: dozer-ingestion/src/connectors/delta_lake/reader.rs (hunk 4 above)
Comment (chloeminkyung) [reply dozer-ingestion/src/connectors/delta_lake/reader.rs:51]: sure

**Pull request merged**